src/idl_parser.cpp

```diff
@@ -897,6 +897,29 @@ CheckedError Parser::ParseTable(const StructDef &struct_def, std::string *value,
     EXPECT(',');
   }
 
+  // Check if all required fields are parsed.
+  for (auto field_it = struct_def.fields.vec.begin();
+            field_it != struct_def.fields.vec.end();
+            ++field_it) {
+    auto required_field = *field_it;
+    if (!required_field->required) {
+      continue;
+    }
+    bool found = false;
+    for (auto pf_it = field_stack_.begin();
```

Comment:
This goes through the entire stack, which means it can possibly find fields in parent objects. You probably want something like `auto pf_it = field_stack_.end() - fieldn` instead?